mne_sandbox/preprocessing/dss.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Apr 19 12:47:10 2016
+@author: drmccloy
+"""
+
+import numpy as np
+
+
+def dss(epochs, data_n_keep=None, bias_n_keep=None, data_thresh=None,
+        bias_thresh=None):
+    """Preprocess physiological data with denoising source separation (DSS)
+
+    Implementation follows the procedure described in S\"arel\"a & Valpola [1]
+    and de Cheveign\'e & Simon [2].
+
+    Parameters
+    ----------
+    data : instance of Epochs
+        Data to be denoised.
+    data_n_keep, bias_n_keep : int
+        Number of components to keep during PCA decomposition of the data and
+        bias function. ``None`` (the default) means keep all suprathreshold
+        components (see ``data_thresh``).
+   data_thresh, bias_thresh : float
+       Threshold (relative to the largest component) below which components
+       will be discarded during decomposition of the data and bias function.
+       For example, if the largest eigenvalue is 3 and ``thresh=0.1``,
+       components with eigenvalues below 0.3 will be discarded.
+
```

Comment:
Tricky question.  DSS will convert channels to (a possibly reduced number of) virtual channels, so returning a modified Epochs instance (whether in-place or copied) will require changing channels / channel names.  In light of that, does it make more sense to return just the DSS data?